src/components/AlertModal/AlertModal.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { FunctionComponent } from "react";
+import { Modal, StyleSheet, Text, View } from "react-native";
+import AlertIcon from "../../../assets/icons/alert.svg";
+import { DarkButton } from "../Layout/Buttons/DarkButton";
+import { SecondaryButton } from "../Layout/Buttons/SecondaryButton";
+import { DangerButton } from "../Layout/Buttons/DangerButton";
+import {
+  borderRadius,
+  color,
+  fontSize,
+  shadow,
+  size
+} from "../../common/styles";
+import { AppText } from "../Layout/AppText";
+
+const styles = StyleSheet.create({
+  centeredView: {
+    flex: 1,
+    width: "100%",
+    justifyContent: "center",
+    alignItems: "center",
+    alignSelf: "center",
+    backgroundColor: "rgba(0,0,0,0.6)"
+  },
+  modalView: {
+    alignItems: "center",
+    marginBottom: 0,
+    paddingTop: size(3),
+    width: 280,
+    backgroundColor: "white",
+    borderRadius: borderRadius(3),
+    ...shadow(1)
+  },
+  alertIcon: {
+    marginBottom: 9
+  },
+  modalTitle: {
+    fontWeight: "bold",
+    fontFamily: "brand-bold",
+    fontSize: fontSize(2),
+    color: color("blue", 50),
+    textAlign: "center",
+    marginHorizontal: 28,
+    marginBottom: size(0.5)
+  },
+  modalText: {
+    marginHorizontal: 32.5,
+    marginBottom: size(4),
+    textAlign: "center"
+  },
+  modalGroupButton: {
+    flexDirection: "row",
+    marginHorizontal: size(2),
+    marginBottom: size(3)
+  },
+  modalSecondaryButton: {
+    marginRight: size(1),
+    flexShrink: 1
+  },
+  modalPrimaryButton: {
+    flex: 1
+  }
+});
+type AlertType = "ERROR" | "WARN" | "CONFIRM" | "INFO";
+export type CallToActionKeyType = "YES_NO" | "OK_CANCEL" | "CONFIRM_CANCEL";
+type CallToActionButtonTexts = {
+  primaryActionText: string;
+  secondaryActionText?: string;
+};
+
+const callToActionCollection: {
+  [key in CallToActionKeyType]: CallToActionButtonTexts;
+} = {
+  YES_NO: { primaryActionText: "YES", secondaryActionText: "NO" },
+  OK_CANCEL: { primaryActionText: "OK", secondaryActionText: "CANCEL" },
+  CONFIRM_CANCEL: {
+    primaryActionText: "CONFIRM",
+    secondaryActionText: "CANCEL"
+  }
+};
+
+export interface AlertModalProps {
+  alertType: AlertType;
+  title: string;
+  description: string;
+  buttonTextType: CallToActionKeyType;
+  visible: boolean;
+  onOk: () => void;
+  onCancel?: () => void;
+  onExit?: () => void;
+}
+
+export const AlertModal: FunctionComponent<AlertModalProps> = ({
+  alertType,
+  title,
+  description,
+  buttonTextType,
+  visible,
+  onOk,
+  onCancel,
+  onExit
+}) => {
+  const primaryButton = (): JSX.Element => {
```

Comment:
Minor nitpick, do we still need this as a function?

Reply:
ok, changed to assignment.